File: 14-longest-common-prefix/longest-common-prefix.ts
function longestCommonPrefix(strs: string[]): string {
    for (let charIndex: number = 0; charIndex < strs[0].length; charIndex++) {
        for (let stringIndex: number = 1; stringIndex < strs.length; stringIndex++) {
            if (
                strs[stringIndex].length <= charIndex ||
                strs[stringIndex][charIndex] !== strs[0][charIndex]
            ) {
                return strs[0].slice(0, charIndex);
            }
        }
    }
    return strs[0];
}

File: 14-longest-common-prefix/longest-common-prefix.go
func longestCommonPrefix(strs []string) string {
    if len(strs) == 0 {
        return ""
    }

    for charIndex := 0; charIndex < len(strs[0]); charIndex++ {
        for stringIndex := 1; stringIndex < len(strs); stringIndex++ {
            // Check if current string is too short or mismatch occurs
            if charIndex >= len(strs[stringIndex]) || strs[stringIndex][charIndex] != strs[0][charIndex] {
                return strs[0][:charIndex]
            }
        }
    }
    return strs[0]
}